references/kvm-admin-api/apiproxy/targets/default.xml

```diff
@@ -0,0 +1,32 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<TargetEndpoint name="default">
+    <Description/>
+    <FaultRules/>
+    <PreFlow name="PreFlow">
+        <Request>
+            <Step>
+                <Name>JS-IAMPermission</Name>
+            </Step>
+            <Step>
+                <Name>AM-IAMPermissionCheck</Name>
+            </Step>
+        </Request>
+        <Response/>
+    </PreFlow>
+    <PostFlow name="PostFlow">
+        <Request/>
+        <Response>
+            <Step>
+                <Name>RF-403</Name>
```

Comment:
This is most likely not enough. The way the `testIAMPermissions` check is implemented it always returns a 200 even if the permission is not available. E.g. with a read only service account I can list the KVMs

```sh
curl -H "Authorization: Bearer $TOKEN" -X POST "https://apigee.googleapis.com/v1/organizations/$APIGEE_ORG/environments/$APIGEE_ENV\:testIamPermissions" -H "Content-Type: application/json" -d "{\"permissions\": [\"apigee.keyvaluemaps.list\"]}"    
{
  "permissions": [
    "apigee.keyvaluemaps.list"
  ]
}
```
but I cannot create them:

```sh
curl -H "Authorization: Bearer $TOKEN" -X POST "https://apigee.googleapis.com/v1/organizations/$APIGEE_ORG/environments/$APIGEE_ENV\:testIamPermissions" -H "Content-Type: application/json" -d "{\"permissions\": [\"apigee.keyvaluemaps.create\"]}"     
{}
```

Both calls return a 200 though.

Reply:
You are right. I did overlook this key aspect. Fixed.